File: apps/api/internal/inventory/repository.go
package inventory

import (
	"context"
	"time"
)

type Repository interface {
	Ping(context.Context) error
	ListItems(context.Context, Filter) ([]Item, error)
	GetItem(context.Context, string) (Item, error)
	CreateItem(context.Context, Item) (Item, error)
	UpdateItem(context.Context, Item) (Item, error)
	ArchiveItem(context.Context, string, time.Time) (Item, error)
	RestoreItem(context.Context, string, time.Time) (Item, error)
	ApplyEvent(context.Context, StockEvent, Item) (Item, error)
	ListEvents(context.Context, string, time.Time) ([]StockEvent, error)
}

File: apps/worker/src/inventory/repository.ts
import type { CommandContext } from "../platform/context";
import type { InventoryFilter, InventoryItem, StockEvent } from "./model";

export interface AuditChange {
  op: "add" | "replace" | "remove";
  path: string;
  value?: unknown;
  oldValue?: unknown;
}

export interface InventoryMutation {
  context: CommandContext;
  action: string;
  expectedVersion: number;
  changes: AuditChange[];
}

export interface InventoryRepository {
  listItems(householdId: string, filter: InventoryFilter): Promise<InventoryItem[]>;
  getItem(householdId: string, itemId: string): Promise<InventoryItem>;
  createItem(item: InventoryItem, mutation: InventoryMutation): Promise<InventoryItem>;
  updateItem(item: InventoryItem, mutation: InventoryMutation): Promise<InventoryItem>;
  setArchived(item: InventoryItem, mutation: InventoryMutation): Promise<InventoryItem>;
  applyStockEvent(item: InventoryItem, event: StockEvent, mutation: InventoryMutation): Promise<InventoryItem>;
  listEvents(householdId: string, itemId: string, since: string): Promise<StockEvent[]>;
}
